client/lib/store-sandbox-helper/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { ToggleControl } from '@wordpress/components';
+import { useState, useEffect } from 'react';
+import ReactDom from 'react-dom';
+import useStoreSandboxStatusQuery from 'calypso/data/store-sandbox/use-store-sandbox-status';
+import wp from 'calypso/lib/wp';
+
+import './style.scss';
+
+interface StoreSandboxQueryResponse {
+	store_sandbox_cookie_name: string | undefined;
+	store_sandbox_expiration_time: number | undefined;
+	is_disabled: boolean | undefined;
+}
+
+export function StoreSandboxHelper() {
+	const { data: isSandboxed = false } = useStoreSandboxStatusQuery();
+	const [ isStoreSandboxed, setIsStoreSandboxed ] = useState( isSandboxed );
+
+	useEffect( () => {
+		setIsStoreSandboxed( isSandboxed );
+	}, [ isSandboxed ] );
+
+	const onToggleStoreSandbox = () => {
+		wp.req.post(
+			{
+				path: `/store-sandbox/${ isStoreSandboxed ? 'disable' : 'enable' }`,
+				apiNamespace: 'wpcom/v2',
+			},
+			{},
+			( error: string, data: StoreSandboxQueryResponse ) => {
+				if ( ! error && data ) {
```

Comment:
If there is an `error`, should we do something to display it or otherwise indicate the request failed?

Reply:
Great suggestion - implemented in ddf9a96ca2e07e7bf142ae0024f45f3ef910b5dd